src/deposit.ts

```diff
@@ -125,18 +124,15 @@ export class DepositFlow extends DepositBaseFlow {
         },
       }); // included in a block on L1
 
-      const l2TxHash = utils.getL2HashFromPriorityOp(
-        txReceipt,
-        await this.wallet._providerL2().getMainContractAddress()
-      );
+      const l2TxHash = utils.getL2HashFromPriorityOp(txReceipt, this.zkChainAddress);
       const txHashs = `(L1: ${depositHandle.hash}, L2: ${l2TxHash})`;
       this.logger.info(`Tx ${txHashs} mined on l1`);
       // wait for deposit to be finalized
       await this.metricRecorder.stepExecution({
         stepName: STEPS.l2_execution,
         stepTimeoutMs: PRIORITY_OP_TIMEOUT,
         fn: async ({ recordStepGasPrice, recordStepGas, recordStepGasCost }) => {
-          const receipt = await depositHandle.wait(1);
+          const receipt = unwrap(await this.l2EthersProvider.waitForTransaction(l2TxHash, 1));
```

Comment:
why not `wallet.provider` avoiding passing extra variable

Reply:
Old provider is from zksync-ethers and when I do handle.wait() it fails because it expects l2ToL1Logs of specific format be present in the receipt. Format has changed in zkos.
I've added comment